File: be-cloud-daakiya-service-release/go-app/campaign/model/campaign.go
package model

import (
	"time"
)

type Campaign struct {
	ID               string                   `json:"id"`
	Name             string                   `json:"name"`
	CreatedAt        time.Time                `json:"createdAt"`
	UpdatedAt        time.Time                `json:"updatedAt"`
	Status           CampaignStatus           `json:"status"`
	CohortIDs        []string                 `json:"campaignCohortIDs"`
	NotificationMeta CampaignNotificationMeta `json:"notificationMeta"`
}

//func main() {
//	// Example JSON string
//	jsonStr := `{
//		"name": "Example Campaign",
//		"campaignCohortIDs": ["cohort1", "cohort2"],
//		"notificationMeta": {
//			"title": "Example Title",
//			"body": "Example Body"
//		}
//	}`
//
//	// Unmarshal the JSON string into a Campaign object
//	var campaign Campaign
//	err := json.Unmarshal([]byte(jsonStr), &campaign)
//	if err != nil {
//		fmt.Println("Error unmarshalling JSON:", err)
//		return
//	}
//
//	// Output the unmarshalled object to the console
//	fmt.Printf("Campaign: %+v\n", campaign)
//}

File: be-cloud-daakiya-service-release/app/src/campaign/models/campaign.ts
import {CampaignNotificationMeta} from './campaign.notification.meta.model';

export enum CampaignStatus {
    INITIATED = 'INITIATED',
    CREATION_IN_PROGRESS = 'CREATION_IN_PROGRESS',
    CREATION_COMPLETE = 'CREATION_COMPLETE',
    SENDING_IN_PROGRESS = 'SENDING_IN_PROGRESS',
    SENDING_COMPLETE = 'SENDING_COMPLETE'
}

export class Campaign {
    id?: string;
    name: string;
    createdAt?: Date;
    updatedAt?: Date;
    status?: CampaignStatus;
    campaignCohortIDs?: string[];
    notificationMeta: CampaignNotificationMeta;

    constructor(data: Campaign) {
        this.id = data.id;
        this.name = data.name;
        this.createdAt = data.createdAt;
        this.updatedAt = data.updatedAt;
        this.status = data.status;
        this.campaignCohortIDs = data.campaignCohortIDs;
        this.notificationMeta = new CampaignNotificationMeta(data.notificationMeta);
    }
}
